modules/javafx.controls/src/main/java/javafx/scene/control/Control.java

```diff
@@ -239,6 +239,13 @@ private static Class<?> loadClass(final String className, final Object instance)
 
         @Override protected void invalidated() {
             Skin<?> skin = get();
+            // check whether the skin is for right control
+            if (skin != null) {
+                if(skin.getSkinnable() != Control.this) {
```

Comment:
minor: Perhaps write this as one `if` (saves an indent level) or at least make the space after `if` consistent between the two.

Reply:
added space.

with your permission, I'd rather have one statement per line - it's easier to step through in a debugger.